app/main.py

```diff
@@ -1,6 +1,35 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cache_storage = dict()
+
+    def add_to_storage(func_name: str, args: tuple) -> int:
+
+        func_result = func(*args)
+        launch_in = str(args)
+        cache_storage[func_name][launch_in] = func_result
+        print("Calculating new result")
+        return func_result
```

Comment:
The function `add_to_storage` is defined inside the `cache` function, which is fine, but it is not reused anywhere else. Consider defining it outside if you plan to use it elsewhere or if it makes the code cleaner.